### DISCUSSION TITLE
Is it possible to remove Dehumidifier from HumidifierDehumidifier accessory ?

### DISCUSSION THREAD

**Discussion by Sukikui** [HomeSpan/HomeSpan#635] [Q&A] [answered]
Hi!

On my home app I've a HumidifierDehumidifier accessory with 4 mode (inactive, auto, hum, dehum). I just want to know if is it possible to remove dehum in order to just keep Humidifier with 2 mode (inactive, hum) ?

Also I've an other question. I'm trying to display some informations about my sensors (accelerometer, angle rate) and I would like to know what accessory (and what combinaison of accessory) is the best option to do that. Right now, I can see that the Thermostat and HumidifierDehumidifier are pretty cool to display information without having to click on their "block" in home app. Temperature sensor are also compact but I failed to display them in a same blocks :/

So if you have a response for my first problem and idea for the second, I take it!
Thanks in advance :)

**Comment by HomeSpan** [comment, marked as answer]
To limit the number of options an Accessory shows for a TargetState Characteristic in the Home App, use the `setValidValues()` method.  As an example of how to restrict to humidifier-only mode, see the HomeSpan [Humidifier/Dehumidifier Reference Sketch](https://github.com/HomeSpan/HomeSpanReferenceSketches/blob/main/Humidifier-Dehumidifier/Humidifier-Dehumidifier.ino). 

Displaying info about sensors is unfortunately hard to do on the main screen in the Home App.  Apple has upgraded the Home App a number of times over the years and now seems to hide most information on the screen behind each tile block.  And as you found, if you have multiple Temperature Sensors, the Home App simply shows them as a single icon with the range of temperatures indicated.

If you can live with having all the data in the secondary screen, the easiest solution is probably to the use Air Quality Sensor since it contains multiple numeric characteristics that you can use to represent alternative data, though you *can't* change the names displayed.

A better possibility would be to use the *Eve for HomeKit App* instead of the Home App.  The Eve App is free in the App store, and is primarily used to control Eve HomeKit Accessories (Eve-brand lights, sensors, etc.), but the interface is in some ways a lot cleaner than the Home App, and it is *generic*, meaning it will render all types of Accessories as long as they meet HAP standards.  From within HomeSpan you can create Custom Characteristics with their own names and add them to a standard HomeKit Service.  The Home App will ignore them, but the Eve App will display them *with the appropriate labeling.*  Alternatively, you can create a complete Custom Service with Custom Characteristics.  A device like this can be paired with HomeKit, though the Home App will complain that the Accessory is "Not Supported."  However, it will be properly displayed in the Eve for HomeKit App.  See the [API](https://github.com/HomeSpan/HomeSpan/blob/master/docs/Reference.md#custom-characteristics-and-custom-services-macros) for details and links to an example sketch.

**Reply by Sukikui** [reply]
Thank you for your very complete answer! I'll try all of that 😄
